u2f-host/devs.c

```diff
@@ -176,15 +177,54 @@ close_devices (u2fh_devs * devs)
   devs->num_devices = 0;
 }
 
+static void
+generate_nonce (unsigned char nonce[8])
+{
+  /*
+   * Generate a nonce.  Cryptographic randomness is nice but not
+   * critical.  The nonce is, in theory, only used to disambiguate
+   * simultaneous requests.
+   */
+
+  uint64_t val;
+
+#ifndef _WIN32  /* _WIN32 includes 64-bit windows. */
+  int fd;
+  int tmp;
+  struct timespec t;
+
+  /* Try /dev/urandom. */
+  fd = open("/dev/urandom", O_RDONLY | O_CLOEXEC);
+  if (fd != -1)
+    {
+      tmp = read(fd, nonce, 8);
+      close(fd);
+      if (tmp == sizeof(nonce))
+	return;
+    }
+
+  if (clock_gettime(CLOCK_MONOTONIC, &t) == 0)
+    {
+      val = t.tv_nsec + (t.tv_sec * 1000000000ULL);
```

Comment:
This is terrible. Don't use CLG! 
Windows has CryptGenRandom function that is euivalent of /dev/urandom  https://msdn.microsoft.com/en-us/library/windows/desktop/aa379942(v=vs.85).aspx